src/main/scala/com/webtrends/harness/component/akkahttp/logging/AccessLog.scala

```diff
@@ -16,59 +16,51 @@
 
 package com.webtrends.harness.component.akkahttp.logging
 
-import akka.http.scaladsl.model.{DateTime, HttpRequest, StatusCode}
-import com.webtrends.harness.command.BaseCommand
-import com.webtrends.harness.component.akkahttp.logging.AccessLog._
+import akka.http.scaladsl.model.{DateTime, StatusCode}
 import com.webtrends.harness.component.akkahttp.routes.AkkaHttpRequest
 import com.webtrends.harness.logging.Logger
 
 object AccessLog  {
   val host: String = java.net.InetAddress.getLocalHost.getHostName
   val accessLog = Logger("AccessLog")
 
-  // Override to obtain the id
-  // id is no longer limited to user, with oauth or other authentication/authorization there are other ids to consider.
-  // Like client id.
-  def getAccessLogId(akkaHttpRequest: AkkaHttpRequest): Option[String] = {
-    None
-  }
-
-  def logAccess(request: HttpRequest, akkaHttpRequest: AkkaHttpRequest, statusCode: Option[StatusCode]) = {
+  def logAccess(request: AkkaHttpRequest, accessLogIdGetter: Option[AkkaHttpRequest => String], statusCode: StatusCode) = {
 
     // modify the logback.xml file to write the "AccessLog" entries to a file without all of the prefix information
-    try {
-      val id: String = getAccessLogId(akkaHttpRequest).getOrElse("-")
-      val status: String = statusCode.map(sc => sc.intValue.toString).getOrElse("-")
-      val responseTimestamp: Long = System.currentTimeMillis()
-      val requestTimestamp: Long = akkaHttpRequest.time
-      val elapsedTime: Long = responseTimestamp - requestTimestamp
-      val requestTime: String = DateTime(requestTimestamp).toIsoDateTimeString()
-      val headers = request.headers
-      val origin = headers.find(header => header.name() == "Origin").map(_.value()).getOrElse("-")
-      val user_agent = headers.find(header => header.name() == "User-Agent").map(_.value()).getOrElse("-")
+    accessLogIdGetter.foreach(getter => {
```

Comment:
Why call this method in the first place if you don't intend to log? Pull ownership of conditional outside of this method.